sds_data_manager/constructs/sqs_construct.py

```diff
@@ -60,7 +60,69 @@ def __init__(
             ),
         )
 
+        # This queue is a special queue that delivers messages after a 15 minute delay
+        # to avoid race conditions between files arriving. Currently, only MAG L1B are
+        # in this queue.
+        self.delay_queue = aws_sqs.Queue(
+            self,
+            "FileArrivalDelayQueue",
+            queue_name="file_arrival_delay_queue.fifo",
+            fifo=True,
+            encryption=aws_sqs.QueueEncryption.UNENCRYPTED,
+            # This timeout determines how long the queue waits for processing. It must
+            # be longer than the timeout of the lambda.
+            visibility_timeout=Duration.seconds(300),
+            # This is required. It removes messages with identical content. Since
+            # the event includes a filename each event should be totally unique.
+            content_based_deduplication=True,
+            # The dead letter queue will take messages that failed retry.
+            dead_letter_queue=aws_sqs.DeadLetterQueue(
+                max_receive_count=1, queue=self.dead_letter_queue
+            ),
+            delivery_delay=Duration.seconds(900),
+        )
+
+        delayed_cases = {"mag": "l1b"}
+
         for instrument_name in instrument_names:
+            instrument_event_match = {
+                "object": {
+                    "key": [{"exists": True}],
+                    "instrument": [instrument_name],
+                },
+            }
+
+            if instrument_name in delayed_cases:
+                data_level = delayed_cases[instrument_name]
+                # Don't send L1B events to the main queue, they go to the delay queue
```

Comment:
Can we rephrase this?
```suggestion
                # Update main queue filters to not send these exception cases. They will go to the delay queue
```